apps/namadillo/src/atoms/integrations/functions.ts

```diff
@@ -213,96 +199,62 @@ const tryCoinToIbcAsset = async (
 
   const originalChainAssets = mapUndefined(assetLookup, originalChainName);
 
-  // TODO: Special handling for Namada because NAM assets are named "tnam...",
-  // and we need to decode this address. For now assuming that if the source
-  // chain is the internal devnet, then the asset is NAM. This is not correct
-  // and should be fixed.
-  if (originalChainName === "internal-devnet-44a.1bd3e6ca62") {
-    const namAsset = originalChainAssets?.[0];
-    if (typeof namAsset === "undefined") {
-      throw new Error(
-        "No NAM asset for internal devnet. This should never happen."
-      );
-    }
-
-    return {
-      asset: namAsset,
-      ...ibcProps,
-    };
-  }
-
   const originalChainRegistryAsset = originalChainAssets?.find(
     (asset) => asset.base === baseDenom
   );
 
-  const assetProp =
-    typeof originalChainRegistryAsset !== "undefined" ?
-      { asset: originalChainRegistryAsset }
-    : unknownAsset(path + "/" + baseDenom);
-
-  return {
-    ...assetProp,
-    ...ibcProps,
-  };
+  return originalChainRegistryAsset || unknownAsset(path + "/" + baseDenom);
 };
 
-const unknownAsset = (denom: string): AssetAndIbc => ({
-  asset: {
-    denom_units: [
-      {
-        denom,
-        exponent: 0,
-      },
-    ],
-    base: denom,
-    name: denom,
-    display: denom,
-    symbol: denom,
-  },
+const unknownAsset = (denom: string): Asset => ({
```

Comment:
Can you check the `apps/namadillo/src/registry/unknownAsset.ts`, maybe we can combine both.

Particularly, I like this idea of having a function which we can at least render the denom instead of `?`

Reply:
That sounds good! I think `unknownAsset.ts` is now not used anywhere. Is there a way you'd prefer to combine them i.e. which properties to keep from each type?

Reply:
cool, no preferences, this is just to render anything for border cases that should never be used